### ISSUE TITLE
[2.0 Beta] Variables inside of shortened Switch statements still included

### ISSUE THREAD

**Issue by SetsuneW** [chrismaltby/gb-studio#532]
**Describe the bug**
Custom Events (and possibly other Events) still consider the Variables inside of shortened Switch statements used, despite not being part of the active script.

**To Reproduce**
1. Create a Custom Event.
2. Create a Switch statement, write code that includes variables.
3. Tell the Switch statement to shorten to before the point where those variables (for example, put a unique variable on the 4th option, then shorten it to 3 options).
4. Though no longer active, the display up-top will continue to show the unused variable as one of your limit of 10.

It's possible this Switch statement behavior exists normally too, just goes unnoticed due to no limits on variable use.

**Expected behavior**
Hopefully the behavior of retaining the contents of a Switch statement you shorten and then lengthen again is still possible, but either way a variable that is not part of the active Switch statement should hopefully not be included in your Variables list or in compiled code.

**Platform (please complete the following information):**
    OS: Windows 10 Home Version 10.0.19041 Build 19041
    App version 2.0.0-beta1, commit 096249d

**Additional context**
I seem to recall some mention of this possibly being in the works to be fixed (variables being included in Commented events), but I wasn't sure and didn't spot a relevant Report, so I posted it here just in case. Apologies if it's a duplicate.

**Comment by RichardULZ** [comment]
Can you test if this happens with a "Disabled" event that uses a variable within a custom event? 
Perhaps this also contributing to custom event variable selection, 
A very confusing issue as this could cause problems with any placed custom events that now have more or less variable inputs defined.

**Comment by SetsuneW** [comment]
Yes, both inside and outside of a Switch statement, with both the Switch statement lengthened to include the event or shortened to hide it, the Variable still shows as being used even though the event is Disabled.

**Comment by pau-tomas** [comment]
Definitely a bug that can become confusing very quickly.

I have a fix for the `disabled` case. However the `Switch` case it's going to be more challenging to fix and it also happens with any `If... Else...` event when disabling the Else path.
